src/genericobject.cpp

```diff
@@ -182,3 +182,15 @@ std::string gob_cmd_update_nametag_attributes(video::SColor color)
 	writeARGB8(os, color);
 	return os.str();
 }
+
+std::string gob_cmd_update_infant(u16 id, u8 type, std::string client_initialization_data)
```

Comment:
change this proto to : 
void gob_cmd_update_infant(const u16 id, const u8 type, const std::string &client_initialization_data, std::string &result)

and write os.str() to result  instead of return os.str()